planner/core/planbuilder.go

```diff
@@ -2931,7 +2931,7 @@ func (b *PlanBuilder) buildAnalyzeAllIndex(as *ast.AnalyzeTableStmt, opts map[as
 				continue
 			}
 
-			p.IdxTasks = append(p.IdxTasks, generateIndexTasks(idx, as, tblInfo, names, physicalIDs, version)...)
+			p.IdxTasks = append(p.IdxTasks, generateIndexTasks(idx, as, as.TableNames[0], names, physicalIDs, version)...)
```

Comment:
A questions from me,
Does the problem only affect global index? But if we change `tblInfo` to `as.TableNames[0]` here, it will also affects normal index.

Am I right?